### PR TITLE
Fix Using

### PR DESCRIPTION
**Pull request by HoshinoTented**

### CODE REVIEW

**Comment by ice1000** [comment]
:tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada:

File: base/src/main/java/org/aya/resolve/context/ModuleExport.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+   * @return false if failed
+   */
+  public boolean export(@NotNull String name, @NotNull AnyVar ref) {
+    var exists = putIfAbsent(name, ref);
+    return exists.isEmpty();
+  }
+
+  // TODO: delete this garbage
+  @Deprecated
+  public @NotNull Result filter(@NotNull ImmutableSeq<String> names, boolean isUse, @NotNull SourcePos pos) {
```
Comment (HoshinoTented) [inline base/src/main/java/org/aya/resolve/context/ModuleExport.java:38]: I hope the `Name`s in UseHide are `WithPos`.

Comment (ice1000) [reply base/src/main/java/org/aya/resolve/context/ModuleExport.java:38]: These names are also serialized, so they do not always have `SourcePos`. Is there another way around?

Comment (ice1000) [reply base/src/main/java/org/aya/resolve/context/ModuleExport.java:38]: But done.

**Review by ice1000 — APPROVED** [review]
bors r+

**Comment by ice1000** [comment]
Thank you!

**Comment by HoshinoTented** [comment]
> Thank you!

I think we need to wait for imkiva's review.

**Comment by ice1000** [comment]
> > Thank you!
> 
> I think we need to wait for @imkiva's review.

We can do follow-up pull requests. If I recall correctly @imkiva is sick yesterday, so if we really want to wait it may take some (more than usual) time

**Pull request merged**